apps/wallet/src/ui/app/components/explorer-link/Explorer.ts

```diff
@@ -9,20 +9,23 @@ const API_ENV_TO_EXPLORER_URL: Record<API_ENV, string | undefined> = {
     [API_ENV.local]: process.env.EXPLORER_URL_LOCAL,
     [API_ENV.devNet]: process.env.EXPLORER_URL_DEV_NET,
     [API_ENV.staging]: process.env.EXPLORER_URL_STAGING,
+    [API_ENV.testNet]: process.env.EXPLORER_URL_TEST_NET,
+    //No explorer url for Custom PRC
+    [API_ENV.customRPC]: '',
 };
 
 // TODO: rewrite this
 function getDefaultUrl(apiEnv?: API_ENV) {
     const url = API_ENV_TO_EXPLORER_URL[apiEnv || DEFAULT_API_ENV];
-    if (!url) {
+    if (!url && !API_ENV.customRPC) {
         throw new Error(`Url for API_ENV ${DEFAULT_API_ENV} is not defined`);
     }
     return url;
 }
 
 export class Explorer {
     public static getObjectUrl(objectID: ObjectId, apiEnv: API_ENV) {
-        return new URL(`/objects/${objectID}`, getDefaultUrl(apiEnv)).href;
+        return new URL(`/objects/${objectID}`, getDefaultUrl(apiEnv))?.href;
```

Comment:
why do why need `?` here?

Reply:
Just missed it.

Reply:
I don't think it's necessary though. new URL either will through or be a URL

Reply:
<img width="472" alt="Screenshot 2022-11-08 at 15 08 35" src="https://user-images.githubusercontent.com/10210143/200601024-1ef69e86-042f-4496-bc13-244745b52eef.png">